pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -21,10 +37,19 @@ def fix_trailing_whitespace(argv=None):
         'grep', '-l', '[[:space:]]$', *args.filenames, retcode=None
     ).strip().splitlines()
 
+    # combine all extension arguments, splitting at ',' and normalizing them
+    # (lowercase and remove unnecessary leading '.' which may be present)
+    md_args = vars(args).get('markdown_linebreak_ext', ['md', 'markdown'])
+    md_exts = [x.lower().lstrip('.') for x in ','.join(md_args).split(',')]
+    all_markdown = '*' in md_exts
+
     if bad_whitespace_files:
         for bad_whitespace_file in bad_whitespace_files:
+            # get extension ([1]); remove leading '.' ([1:]); and make lowercase
+            extension = os.path.splitext(bad_whitespace_file)[1][1:].lower()
```

Comment:
another idea, since in both places you're normalizing to not-dot, you could instead normalize to dotted-always in just one place (might avoid the no-extension case.  If that's the case I'd suggest changing this code to:

```
_, extension = ...splitext(...)
```

which is slightly less error prone and slightly more self documenting